src/App.tsx

```diff
@@ -1,14 +1,35 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const selectedTodo = todos.find(todo => todo.id === selectedTodoId) || null;
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    getTodos().then((response) => {
```

Comment:
Let’s use `async await`. It gives you a simpler way to work with asynchronous promise-based code.